### PR TITLE
Update using Skeleton2D documentation

### PR DESCRIPTION
**Pull request by ShatReal**
See #6388

I have renamed the cutout animation page to "2D Skeletons" and the 2d skeletons page to "2d Skeletal Deform" and updated the contents of each page accordingly. I cut out the stuff about making bones with sprites because that's not possible anymore.

### CODE REVIEW

File: tutorials/animation/2d_skeletons.rst [hunk 1]
```diff
@@ -3,238 +3,260 @@
...
 
-* Better integration with the engine, so less hassle importing and editing from
-  an external tool.
-* Ability to control particle systems, shaders, sounds, call scripts, colors,
-  transparency, etc. in animations.
-* The built-in skeletal system in Godot is very efficient and designed for
-  performance.
+-  **The animation system is fully integrated with the engine**: This
+   means animations can control much more than just motion of objects. Textures,
+   sprite sizes, pivots, opacity, color modulation, and more, can all be animated
```
Comment (LeoDog896) [inline tutorials/animation/2d_skeletons.rst:33]: You mention pivots, but they seem to be missing from this documentation - [they were covered in the original documentation](https://docs.godotengine.org/en/stable/tutorials/animation/cutout_animation.html).

**Review by LeoDog896 — CHANGES_REQUESTED** [review]

File: tutorials/animation/2d_skeletons.rst (hunk 1 above)
Comment (ShatReal) [reply tutorials/animation/2d_skeletons.rst:33]: Oh right pivots are not necessary any more i don't think so, i'll remove it

Comment (ShatReal) [reply tutorials/animation/2d_skeletons.rst]: okay i did it

**Comment by clayjohn** [comment]
I'm not sure the names are very helpful. The Cutout animation tutorial is more about that specific way of animating sprites than it is about using Skeleton2D. The Skeleton2D tutorial, on the other hand, provides much more detail about using the Skeleton2D node.

**Comment by ShatReal** [comment]
So, should I move the information about Skeleton2D from the  "2D skeletons" page to the "2d skeletal deform" page?

**Comment by fguillen** [comment]
I would love to have some updated documentation on Skeleton2D

File: tutorials/animation/2d_skeletal_deform.rst [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+.. warning::
+
+    There are known issues with 2D skeletons on mobile and web platforms with the GLES2 renderer. We
+    recommend using the GLES3 renderer if your project relies on Skeleton2D for now.
+
+Introduction
+------------
+
+When working with 3D, skeletal deforms are common for characters and creatures
```
Comment (pcholt) [inline tutorials/animation/2d_skeletal_deform.rst:9]: Sometimes you use the term "skeletal deforms" and sometimes "skeletal deformations". I would prefer "skeletal deformations"

File: tutorials/animation/2d_skeletons.rst [hunk 3]
```diff
@@ -1,240 +1,376 @@
...
+Get your assets: :download:`gbot_resources.zip <files/gbot_resources.zip>`.
 
-.. image:: img/gBot_pieces.png
+Setting up the rig
+~~~~~~~~~~~~~~~~~~
 
-It is also advised to download the final character image
-:download:`gBot_complete.png <img/gBot_complete.png>` to have a good reference
-for putting the different pieces together.
+Create an empty Node2D as root of the scene, we will work under it:
```
Comment (pcholt) [inline tutorials/animation/2d_skeletons.rst]: Starting from here, the names for the bone and sprite nodes are different from the defaults, but there are no instructions to change the name of the nodes. How about

> Create an empty Node2D as the root of the scene, and rename it as "GBot". We will work under it:

File: tutorials/animation/2d_skeletons.rst [hunk 4]
```diff
@@ -3,233 +3,260 @@
...
+rotating the body bone. We can do this be pressing :kbd:`E` to enter rotate mode,
+and dragging with the left mouse button. To exit rotate mode hit :kbd:`ESC`.
 
-.. image:: img/skel2d3.png
+.. image:: img/tutovec_torso1.gif
 
-Head over to the *Points* mode, select the pencil and draw a polygon around the
-desired piece:
+Continue adding body pieces, starting with the
+right arm (your left). Make sure to put each sprite in its correct place in the hierarchy,
```
Comment (pcholt) [inline tutorials/animation/2d_skeletons.rst:92]: Very nice 😄

File: tutorials/animation/2d_skeletons.rst [hunk 5]
```diff
@@ -3,233 +3,260 @@
...
+.. image:: img/tuto_cutout3.png
 
-Drawing a polygon directly is not advised. Instead, open the "UV" dialog for the
-polygon:
+This looks good. Let's see if our hierarchy works as a skeleton by
+rotating the body bone. We can do this be pressing :kbd:`E` to enter rotate mode,
+and dragging with the left mouse button. To exit rotate mode hit :kbd:`ESC`.
 
-.. image:: img/skel2d3.png
+.. image:: img/tutovec_torso1.gif
```
Comment (pcholt) [inline tutorials/animation/2d_skeletons.rst:89]: The image here shows the body bone with a Y position transform not mentioned in the text

File: tutorials/animation/img/tuto_cutout7.png
Comment (pcholt) [inline tutorials/animation/img/tuto_cutout7.png:1]: This looks like the same illustration as tuto_cutout6.png, and doesn't match the description text.

**Comment by myyk** [comment]
Hi, I'm new to Godot (and game dev in general). I was following the docs https://docs.godotengine.org/en/stable/tutorials/animation/cutout_animation.html#doc-cutout-animation and it's really confusing how it's so outdated. None of it's talk about how bones work is correct.

I think an even imperfect version of fixed docs would be much more usable than what's there today. I'd love to see this revived and shipped. I'll try to follow instructions from the .rst files.

**Pull request closed without merging**